core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -5491,18 +5491,24 @@ ImmutableList<RelNode> retrieveCursors() {
               builder.add(convertExpression(node));
             }
             final ImmutableList<RexNode> list = builder.build();
+            RelNode rel = root.rel;
+            CorrelationUse correlationUse = getCorrelationUse(this, root.rel);
```

Comment:
Could you add a comment explaining what happens here?

Reply:
Sure, I can add a comment here, but I believe this function/pattern is used elsewhere so thought adding only here might not be  relevant.

Reply:
I am not familiar with this part of the code, so I need help to understand why this works.
Maybe other reviewers won't need this help.